autonomous/autonomous.py

```diff
@@ -28,98 +48,85 @@ class AutoBase(AutonomousStateMachine):
         self.look_around = 0.3
+        self.cur_waypoint = 0
 
-        self.last_pose = self.waypoints[0]
+        self.last_pose = self.waypoints[0].pose
         self.trap_profile_start_time = 0
 
         wpilib.SmartDashboard.putNumber("auto_vel", 0.0)
 
     def setup(self):
-        field_bballs = self.field.getObject("bballs")
-        field_rballs = self.field.getObject("rballs")
         field_goal = self.field.getObject("goal")
-        field_bballs.setPoses(
-            [trajectory_generator.goal_to_field(Pose2d(*b, 0)) for b in blue_balls]
-        )
-        field_rballs.setPoses(
-            [trajectory_generator.goal_to_field(Pose2d(*b, 0)) for b in red_balls]
-        )
         field_goal.setPose(trajectory_generator.goal_to_field(Pose2d(0, 0, 0)))
 
     def on_enable(self):
-        self.target_estimator.set_pose(self.waypoints[0])
+        self.target_estimator.set_pose(self.waypoints[0].pose)
 
-        self.last_pose = self.waypoints[0]
+        self.last_pose = self.waypoints[0].pose
         # generates initial velocity profile
-        self.trap_profile, self.stop_point = self.generate_trap_profile(
-            len(self.waypoints), TrapezoidProfile.State(0, 0)
-        )
+        self.cur_waypoint = 0
+        self.trap_profile = self._generate_trap_profile(TrapezoidProfile.State(0, 0))
         return super().on_enable()
 
-    def generate_trap_profile(
-        self, goal: int, current_state: TrapezoidProfile.State
-    ) -> Tuple[TrapezoidProfile, float]:
-        """Generates a linear trapazoidal trajectory that goes from current state to goal"""
-        stop_point = trajectory_generator.total_length(self.waypoints[:goal])
-        ret = (
-            TrapezoidProfile(
-                self.linear_constraints,
-                goal=TrapezoidProfile.State(stop_point, 0),
-                initial=current_state,
-            ),
-            stop_point,
-        )
-        self.logger.info(f"generated {ret[0].totalTime()}s")
-        return ret
-
-    @state(first=True)
+    @state
     def move(self, tm):
         # indexer controller will hanle it self raising and lowering
         self.indexer_control.wants_to_intake = True
         # always be trying to fire
         self.shooter_control.wants_to_fire = True
         # calculate speed and position from current trapazoidal profile
-        trap_time = tm
+        trap_time = tm - self.trap_profile_start_time
         linear_state = self.trap_profile.calculate(trap_time)
         # TODO: change to our error from final position is below value
         is_done = self.trap_profile.isFinished(trap_time)
 
         if is_done:
-            self.logger.info(f"Done at {tm}")
-            self.next_state("stopped")
+            self.logger.info(f"Got to waypoint{self.cur_waypoint} at {tm}")
+            waypoint_type = self.waypoints[self.cur_waypoint].type
+            if waypoint_type == WaypointType.SHOOT:
```

Comment:
nit: this is an enum we define, so the members are singletons
```suggestion
            if waypoint_type is WaypointType.SHOOT:
```